File: GJOH6/src/task2/model/PhoneContact.java
package task2.model;

import task2.model.enums.Tag;

import java.util.HashMap;
import java.util.Map;

/**
 * The class represents a contact in the phone book
 */
public class PhoneContact extends Contact {

    private final Map<String, Tag> phones;

    public PhoneContact(String firstname, String lastname, String phone, Tag tag) {
        super(firstname, lastname);
        this.phones = new HashMap<>();
        this.phones.put(phone, tag);
    }

    public Map<String, Tag> getPhones() {
        return phones;
    }

    @Override
    public String toString() {
        return getFirstname() + " " + getLastname() + " " + phones;
    }
}

File: GJOH7/model/phone_contact.py
from model.contact import Contact


class PhoneContact(Contact):
    """
    The class represents a contact in the phone book
    """
    def __init__(self, firstname, lastname, numbers=None):
        super().__init__(firstname, lastname)
        self.numbers = numbers if numbers else []

    def add_number(self, number, tag):
        self.numbers.append({'number': number, 'tag': tag})

    def get_phones(self):
        return self.numbers

    def __str__(self):
        return "{} {} {}".format(self.get_firstname(), self.get_lastname(), self.numbers)
